ipa-core/src/protocol/boolean/step.rs

```diff
@@ -0,0 +1,69 @@
+use ipa_step_derive::CompactStep;
+
+#[derive(CompactStep)]
+pub(crate) enum BoolAndStep {
+    #[step(count = 8)] // keep in sync with MAX_BITS defined inside and.rs
+    Bit(usize),
+}
+
+#[derive(CompactStep)]
+pub enum EightBitStep {
+    #[step(count = 8)]
+    Bit(usize),
+}
+
+impl From<usize> for EightBitStep {
+    fn from(v: usize) -> Self {
+        Self::Bit(v)
+    }
+}
+
+#[derive(CompactStep)]
+pub enum SixteenBitStep {
+    #[step(count = 16)]
+    Bit(usize),
+}
+
+impl From<usize> for SixteenBitStep {
+    fn from(v: usize) -> Self {
+        Self::Bit(v)
+    }
+}
+
+#[derive(CompactStep)]
+pub enum ThirtyTwoBitStep {
+    #[step(count = 32)]
+    Bit(usize),
+}
+
+impl From<usize> for ThirtyTwoBitStep {
+    fn from(v: usize) -> Self {
+        Self::Bit(v)
+    }
+}
+
+#[derive(CompactStep)]
+pub enum TwoHundredFiftySixBitOpStep {
+    #[step(count = 256)]
+    Bit(usize),
+}
+
+impl From<usize> for TwoHundredFiftySixBitOpStep {
+    fn from(v: usize) -> Self {
+        Self::Bit(v)
+    }
+}
+
+#[cfg(test)]
+#[derive(CompactStep)]
```

Comment:
I guess we need to look into what the codecov warnings mean here.  Maybe we need to suppress them.